harvest/harvest/pipelines.py

```diff
@@ -4,10 +4,28 @@
 # See: https://docs.scrapy.org/en/latest/topics/item-pipeline.html
 
 
+import requests
+
 # useful for handling different item types with a single interface
 from itemadapter import ItemAdapter  # noqa: F401
 
 
 class HarvestPipeline:
+    def __indico(self):
+        self.indico_id = self.event_details.split("/")[4]
+        r = requests.get(
+            "https://indico.cern.ch/export/event/{}.json".format(self.indico_id)
+        )
+        res_data = r.json()
+
+        sponsor = res_data["results"][0]["organizer"].split(" / \\ - ")[0]
+        keywords = res_data["results"][0]["keywords"]
+
+        return sponsor, keywords
+
     def process_item(self, item, spider):
+        sponsor, keywords = self.__indico()
+        item["sponsor"] = sponsor
+        item["keywords"] = keywords
+
```

Comment:
Some events don't have keywords, for [example](https://indico.cern.ch/export/event/1052076.json) that means that the code will fail. We can do something like the following:

```suggestion
    # on the top of the file
    import backoff
    import logging
    
    LOGGER = logging.getLogger(__name__)
    
   @backoff.on_exception(
        backoff.expo, requests.exceptions.RequestException, max_tries=5
    )
    def __indico(self, indico_id):
        response = requests.get(
            f"https://indico.cern.ch/export/event/{indico_id}.json"
        )
        
        response.raise_for_status()
        data = response.json()
        
        sponsor = None
        try:
            sponsor = data["results"][0]["organizer"].split("/")[0].strip()
        except Exception:
            LOGGER.error(f'Cannot get sponsor for indico event: {indico_id}')
        
        keywords = None
        try:
            keywords = data["results"][0]["keywords"]
        except Exception:
            LOGGER.error(f'Cannot get keywords for indico event: {indico_id}')
        return sponsor, keywords

    def process_item(self, item, spider):
         indico_id = item.get('event_details')
         if indico_id:
             item["sponsor"],  item["keywords"] = self.__indico(indico_id)
         return item
```

Reply:
Noted, changed.